cpp/include/cudf/join.hpp

```diff
@@ -469,20 +461,19 @@ class hash_join {
     rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref()) const;
 
  private:
-  const std::unique_ptr<impl_type const> _impl;
+  std::unique_ptr<impl_type const> _impl;
```

Comment:
Avoid const data member per https://quuxplusone.github.io/blog/2022/01/23/dont-const-all-the-things/#data-members-never-const